### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by KovalevichTimofei**
In this pull request commits I:
1) Finished main functionality of listeners page: create, edit and update information with dynamic update of the interface.
2) Developed mixin for notifications for capability of injecting notifications into each page.
3) Injected notifications into the preachers page. Notifications appear when user creates, updates or deletes any information and log pending, successful and unsuccessful results of the above actions.

### CODE REVIEW

File: src/pages/Listeners.vue [hunk 1]
```diff
@@ -44,24 +96,42 @@ export default {
...
+      id: '',
+      email: '',
+      name: '',
+      surname: '',
     };
   },
   computed: {
     ...mapState({
       listenersList: state => state.listeners.listeners,
+      listener: state => state.listeners.listener,
```
Comment (vasaeps3) [inline src/pages/Listeners.vue]: have to use [mapState ](https://vuex.vuejs.org/api/#mapstate)

File: src/pages/Listeners.vue [hunk 2]
```diff
@@ -44,24 +96,42 @@ export default {
...
       loading: state => state.listeners.loading,
       seminarsList: state => state.seminars.seminars.map(seminar => ({
```
Comment (vasaeps3) [inline src/pages/Listeners.vue]: have to use [mapGetters](https://vuex.vuejs.org/api/#mapgetters)

File: src/pages/Listeners.vue [hunk 3]
```diff
@@ -70,21 +140,71 @@ export default {
       });
     },
     filterListeners(val) {
-      this.$store.dispatch('listeners/fetchAllUsers', {
+      console.log(val);
+      this.$store.dispatch('listeners/fetchAllListeners', {
```
Comment (vasaeps3) [inline src/pages/Listeners.vue]: have to use [mapActions](https://vuex.vuejs.org/api/#mapactions)

File: src/pages/Listeners.vue [hunk 4]
```diff
@@ -70,21 +140,71 @@ export default {
...
       });
     },
+    openConfirmDeleteModal() {
+      this.isConfirmDeleteModalOpen = true;
+    },
+    showCreateModal() {
+      this.isCreateModalOpen = true;
+    },
+    async showEditModal(id) {
+      await this.$store.dispatch('listeners/fetchListener', id);
```
Comment (vasaeps3) [inline src/pages/Listeners.vue]: have to use [mapActions](https://vuex.vuejs.org/api/#mapactions)

**Comment by vasaeps3** [comment]
You shouldn't use in a component this.$store.***
You have to use [binding helpers](https://vuex.vuejs.org/api/#component-binding-helpers)

File: src/pages/Listeners.vue (hunk 1 above)
Comment (KovalevichTimofei) [reply src/pages/Listeners.vue]: Done.

File: src/pages/Listeners.vue (hunk 2 above)
Comment (KovalevichTimofei) [reply src/pages/Listeners.vue]: Done.

File: src/pages/Listeners.vue (hunk 3 above)
Comment (KovalevichTimofei) [reply src/pages/Listeners.vue]: Done.

File: src/pages/Listeners.vue (hunk 4 above)
Comment (KovalevichTimofei) [reply src/pages/Listeners.vue]: Done.

**Comment by KovalevichTimofei** [comment]
Done.

**Pull request merged**